### DISCUSSION TITLE
Networking Question

### DISCUSSION THREAD

**Discussion by akoenig** [coder/coder#6627] [General] [answered]
I'm currently wrapping my head around if it is secure to not expose the `code-server` via `https` when operating the following network structure:

![image](https://user-images.githubusercontent.com/224910/225429917-60c138b8-7992-4db6-a1cb-2c53509f931a.png)

- All instances are running within a public cloud.
- All instances are grouped in an own Tailnet.
- The Coder Server is only accessible when connected to the Tailnet via the `ts.net` domain.
- The `code-server` running on the machines is exposed via port 8080 (unencrypted, http).

I'm just wondering if this is still a secure way to operate the `code-server` on the machines or if we should also establish a self-signed cert here? All instances are also shielded by security groups / firewall rules. Is there (in any way) still a possibility to access the `code-server` when not connected to the Tailnet (e.g. via the public IP address of the machine)?

**Comment by matifali** [comment]
@bpmct if you comment here.

**Reply by bpmct** [reply, marked as answer]
Hey - I think we chatted on Discord about this. Running code-server on 8080 with no auth (HTTP) is still safe from Coder's perspective, since it'll be reverse-proxied with TLS.
